builder/slide_and_fold_storage.cc

```diff
@@ -377,7 +377,19 @@ class slide_and_fold : public node_mutator {
         // to move the loop min back so we compute the whole required region.
         expr new_min_at_new_loop_min = substitute(new_min, loop.sym, x);
         expr old_min_at_loop_min = substitute(old_min, loop.sym, loop.bounds.min);
-        expr new_loop_min = where_true(new_min_at_new_loop_min <= old_min_at_loop_min, x).max;
+        expr new_loop_min = negative_infinity();
+
+        // TODO: this search can be more efficient by trying to cover wider range of depth 
+        // using binary search or something similar.
+        const int max_search_depth = 10;
```

Comment:
Can we put this in a little helper function like `where_true`? This is basically just `where_true` with an initial guess for the upper bound.

Reply:
Done.